controller/engine_controller.go

```diff
@@ -663,10 +661,32 @@ func (m *EngineMonitor) refresh(engine *longhorn.Engine) error {
 	}
 	if !reflect.DeepEqual(engine.Status.ReplicaModeMap, currentReplicaModeMap) {
 		engine.Status.ReplicaModeMap = currentReplicaModeMap
-		_, err = m.ds.UpdateEngine(engine)
+	}
+
+	info, err := client.Info()
+	if err != nil {
 		return err
 	}
-	return nil
+	// for those engine just restored from backup, field LastRestored is empty
+	if info.LastRestored != "" {
+		engine.Status.LastRestoredBackup = info.LastRestored
+	}
+
+	if engine.Spec.RquestedBackupRestore != "" && engine.Spec.RquestedBackupRestore != engine.Status.LastRestoredBackup {
+		backupTarget, err := manager.GenerateBackupTarget(m.ds)
+		if err != nil {
+			return err
+		}
+		logrus.Infof("engine %v prepared to do incremental restore, backup target %v, backup volume %v, backup name %v",
+			engine.Name, backupTarget.URL, engine.Spec.BackupVolume, engine.Spec.RquestedBackupRestore)
+		err = client.BackupRestoreIncrementally(backupTarget.URL, engine.Spec.RquestedBackupRestore, engine.Spec.BackupVolume, engine.Status.LastRestoredBackup)
```

Comment:
Can we put this in a separate thread? This is a long operation and will block the monitor. So if one replica is down or something else happened, we won't able to deal with it.